velox/dwio/dwrf/writer/Writer.cpp

```diff
@@ -775,17 +775,27 @@ dwrf::WriterOptions getDwrfOptions(const dwio::common::WriterOptions& options) {
         Config::COMPRESSION.configKey(),
         std::to_string(options.compressionKind.value()));
   }
-
+  if (options.minCompressionSize.has_value()) {
+    configs.emplace(
+        Config::COMPRESSION_BLOCK_SIZE_MIN.configKey(),
+        std::to_string(options.minCompressionSize.value()));
+  }
   if (options.maxStripeSize.has_value()) {
     configs.emplace(
         Config::STRIPE_SIZE.configKey(),
         std::to_string(options.maxStripeSize.value()));
   }
+  if (options.linearStripeSizeHeuristics.has_value()) {
+    configs.emplace(
+        Config::LINEAR_STRIPE_SIZE_HEURISTICS.configKey(),
+        std::to_string(options.linearStripeSizeHeuristics.value()));
+  }
   if (options.maxDictionaryMemory.has_value()) {
     configs.emplace(
         Config::MAX_DICTIONARY_SIZE.configKey(),
         std::to_string(options.maxDictionaryMemory.value()));
   }
+
   dwrf::WriterOptions dwrfOptions;
   dwrfOptions.config = Config::fromMap(configs);
```

Comment:
Please use the setters of the Config class directly where possible, instead of emplacing into the config map and serialize from it.

Reply:
Do that in followup as a separate refactor?